[Bug 768] Keep auth mode unchanged after iwconfig key off/on.

Signed-off-by: Zhu Yi <yi.zhu@intel.com>
--
diff -urp ieee80211-1.0.3/ieee80211_wx.c ieee80211-1.0.3-keep_restricted_bit/ieee80211_wx.c
--- ieee80211-1.0.3/ieee80211_wx.c	2005-07-15 07:53:35.000000000 +0800
+++ ieee80211-1.0.3-keep_restricted_bit/ieee80211_wx.c	2005-09-01 13:03:16.000000000 +0800
@@ -413,11 +413,15 @@ int ieee80211_wx_set_encode(struct ieee8
 			sec.flags |= SEC_ACTIVE_KEY;
 		}
 	}
-	ieee->open_wep = !(erq->flags & IW_ENCODE_RESTRICTED);
-	sec.auth_mode = ieee->open_wep ? WLAN_AUTH_OPEN : WLAN_AUTH_SHARED_KEY;
-	sec.flags |= SEC_AUTH_MODE;
-	IEEE80211_DEBUG_WX("Auth: %s\n", sec.auth_mode == WLAN_AUTH_OPEN ?
-			   "OPEN" : "SHARED KEY");
+	if (erq->flags & (IW_ENCODE_OPEN | IW_ENCODE_RESTRICTED)) {
+		ieee->open_wep = !(erq->flags & IW_ENCODE_RESTRICTED);
+		sec.auth_mode = ieee->open_wep ? WLAN_AUTH_OPEN :
+						 WLAN_AUTH_SHARED_KEY;
+		sec.flags |= SEC_AUTH_MODE;
+		IEEE80211_DEBUG_WX("Auth: %s\n",
+				   sec.auth_mode == WLAN_AUTH_OPEN ?
+			   	   "OPEN" : "SHARED KEY");
+	}
 
 	/* For now we just support WEP, so only set that security level...
 	 * TODO: When WPA is added this is one place that needs to change */
